common/prque/prque.go

```diff
@@ -22,62 +22,59 @@ import (
 )
 
 // Priority queue data structure.
-type Prque struct {
-	cont *sstack
+type Prque[V any] struct {
+	cont *sstack[V]
```

Comment:
I think we could have parameters `P comparable, V any` here. That way, one could use any integer type as key.

Reply:
LES uses some weird "wrap around" notion for the priorities, which requires subtracting one priority from another. @zsfelfoldi said that that is a temporary hack 2 years ago :P Until that's fixed, I can't make the priority "comparable" only.

Reply:
This would be the code to make the priority generic too: https://github.com/ethereum/go-ethereum/pull/26290, but it changes LES in - a probably silently - breaking way. We need a fix / confirm frmo @zsfelfoldi to make this change.